payment/datastore.go

```diff
@@ -403,6 +406,77 @@ func (pg *Postgres) IsStripeSub(orderID uuid.UUID) (bool, string, error) {
 	return ok, "", err
 }
 
+// RenewOrder updates the orders status to paid and paid at time, inserts record of this order
+// 	Status should either be one of pending, paid, fulfilled, or canceled.
+func (pg *Postgres) RenewOrder(ctx context.Context, orderID uuid.UUID) error {
+	// create tx
+	tx, err := pg.RawDB().BeginTxx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer pg.RollbackTx(tx)
```

Comment:
Does this only execute if `err != nil`?

Reply:
It executes when the function ends.  There isn't a penalty for rolling back a committed transaction, so this is like a catch all, if for any reason (including panic) this function returns we will rollback, unless we explicitly call commit.